### PR TITLE
added data reduction algorithm info

### PR DESCRIPTION
**Pull request by mvgaliev**

### CODE REVIEW

File: Capabilities/DataViewMappings.md [hunk 1]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
 
 The data can be aggregated by selecting the desired field and clicking sum.  
 ![](images/DataAggregation.png)
+
+# Data Reduction Algorithm
+
+A `DataReductionAlgorithm` can be used if you want to change amount of data receiving by DataView.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: A `DataReductionAlgorithm` can be applied if you want to control the amount of data received in the DataView.

File: Capabilities/DataViewMappings.md [hunk 2]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
 
 The data can be aggregated by selecting the desired field and clicking sum.  
 ![](images/DataAggregation.png)
+
+# Data Reduction Algorithm
+
+A `DataReductionAlgorithm` can be used if you want to change amount of data receiving by DataView. 
+
+By default all Custom Visuals have `top` `DataReductionAlgorithm` with 1000 count just like if we add the following code to capabilities.json:
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: By default all Custom Visuals have **the** `top` `DataReductionAlgorithm` applied with the 'count' set to 1000 dataPoints. This is equivalent to setting the following properties in the capabilities.json:

File: Capabilities/DataViewMappings.md [hunk 3]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+```json
+"dataReductionAlgorithm": {
+    "top": {
+        "count": 1000
+    }
+}
+```
+However you can change this value from 1 to a maximum 30000.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: You can modify the 'count' value to any integer value up to 30000.

File: Capabilities/DataViewMappings.md [hunk 4]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+    "top": {
+        "count": 1000
+    }
+}
+```
+However you can change this value from 1 to a maximum 30000. 
+
+## Data Reduction Algorithm types 
+
+There are 4 types of `DataReductionAlgorithm`:
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: There are 4 types of `DataReductionAlgorithm` settings:

File: Capabilities/DataViewMappings.md [hunk 5]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+* `top` - if you want reduce the data to the top. First "count" values will be taken from a data set.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: if you want limit the data to values taken from the top of the datavset. The top first "count" values will be taken from the data set.

File: Capabilities/DataViewMappings.md [hunk 6]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+* `bottom` - will reduce the data to the bottom. Last "count" of items will be taken instead.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: if you want limit the data to values taken from the bottom of the dataset. The last "count" values will be taken from the data set.

File: Capabilities/DataViewMappings.md [hunk 7]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+* `sample` - reduce the data using a simple Sample of count items.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: * `sample` - means reducing the dataset by a simple sampling algorithm limited to a "count" number of items.

File: Capabilities/DataViewMappings.md [hunk 8]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+```
+However you can change this value from 1 to a maximum 30000. 
+
+## Data Reduction Algorithm types 
+
+There are 4 types of `DataReductionAlgorithm`:
+* `top` - if you want reduce the data to the top. First "count" values will be taken from a data set.
+* `bottom` - will reduce the data to the bottom. Last "count" of items will be taken instead.
+* `sample` - reduce the data using a simple Sample of count items. 
+It means first and last items will be taken. Also "count" of items will be taken from equal intervals between first and last element.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: It means first and last items are included, as well as a "count" number of items having equal intervals between them.

File: Capabilities/DataViewMappings.md [hunk 9]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+For example if you got a data set [0, 1, 2, ... 100] and a `count: 9` then you will receive next values [0, 10, 20 ... 100]
+* `window` - allow the data to be loaded one window, containing count items, at a time
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: `window` - loads one 'window' of datapoints at a time containing 'count' elements.

Q: how do you control which 'window' is loaded?

File: Capabilities/DataViewMappings.md [hunk 10]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+
+## Data Reduction Algorithm usage
+
+`DataReductionAlgorithm` can be used in categorical, table or matrix data mapping.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: categorical, table or matrix **dataview** mapping.

File: Capabilities/DataViewMappings.md [hunk 11]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+
+For categorical data mapping it can be set into `categories` or/and group section of `values`.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: **and\or**

File: Capabilities/DataViewMappings.md [hunk 12]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+                        "count": 100
+                    }
+                }  
+            }
+        }
+    }
+}
+```
+
+For table data mapping data reduction algorithm can be applied to `rows` section.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: dataview mapping

Comment (AviSander) [inline Capabilities/DataViewMappings.md]: to **the** 'rows' section

File: Capabilities/DataViewMappings.md [hunk 13]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+                        "count": 2000
+                    }
+                } 
+            }
+        }
+    }
+]
+```
+
+For `matrix` data mapping data reduction algorithm can be applied to `rows` or/and `columns` section.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: dataview mapping

Comment (AviSander) [inline Capabilities/DataViewMappings.md]: and\or

Comment (AviSander) [inline Capabilities/DataViewMappings.md]: to **the** 'rows

File: Capabilities/DataViewMappings.md (hunk 3 above)
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: R visuals can support up to 150000 rows

File: Capabilities/DataViewMappings.md (hunk 5 above)
Comment (ABKimmel) [inline Capabilities/DataViewMappings.md]: >if you want **to** limit the data

I would also say:
>The **first `count`** values will be taken from the **top of the** data set.

File: Capabilities/DataViewMappings.md (hunk 6 above)
Comment (ABKimmel) [inline Capabilities/DataViewMappings.md]: >if you want **to** limit the data

File: Capabilities/DataViewMappings.md (hunk 7 above)
Comment (ABKimmel) [inline Capabilities/DataViewMappings.md]: Note that the sampling algorithm is: first and last values, plus count values evenly spaced between them. For example if my data set is [1, 2, ..., 99, 100], and count is set to `9`, I would get the data set: [1,11,21,31,41,51,60,70,80,90,100]

File: Capabilities/DataViewMappings.md (hunk 8 above)
Comment (ABKimmel) [inline Capabilities/DataViewMappings.md]: Might want to include an example

File: Capabilities/DataViewMappings.md [hunk 14]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+You can modify the 'count' value to any integer value up to 30000. R-based custom visuals can support up to 150000 rows.
+
+## Data Reduction Algorithm types 
+
+There are four types of `DataReductionAlgorithm` settings::
+* `top` - if you want limit the data to values taken from the top of the dataset. The top first "count" values will be taken from the data set.
+* `bottom` - if you want limit the data to values taken from the bottom of the dataset. The last "count" values will be taken from the data set.
+* `sample` - reduce the dataset by a simple sampling algorithm limited to a "count" number of items. 
+It means first and last items are included, as well as a "count" number of items having equal intervals between them.
+For example if you got a data set [0, 1, 2, ... 100] and a `count: 9` then you will receive next values [0, 10, 20 ... 100]
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: then you will receive the **following** values

File: Capabilities/DataViewMappings.md [hunk 15]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+* `window` - loads one 'window' of datapoints at a time containing "count" elements. Currently `top` and `window` are equivalent. There is work in progress to support this setting.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: There is work in progress to fully support a **windowing** setting

File: Capabilities/DataViewMappings.md [hunk 16]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+                        "count": 100
+                    }
+                }  
+            }
+        }
+    }
+}
+```
+
+For table dataview mapping data reduction algorithm can be applied to the `rows` section.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: For table dataview mapping **the** data reduction

File: Capabilities/DataViewMappings.md [hunk 17]
```diff
@@ -256,3 +256,89 @@ PowerBI will produce you this as the table dataview. Do not assume there is an o
...
+                        "count": 2000
+                    }
+                } 
+            }
+        }
+    }
+]
+```
+
+For `matrix` dataview mapping data reduction algorithm can be applied to the `rows` and/or `columns` section.
```
Comment (AviSander) [inline Capabilities/DataViewMappings.md]: dataview mapping **the** data reduction algorithm

**Pull request merged**